### PR TITLE
Feature/Improve remote tests

### LINKED ISSUE #1152: Remote: Tests needs updating

### PR DESCRIPTION
**Pull request by dcblundell**
Adds some data validation and expands remote tests.

Closes https://github.com/tenstorrent/ttnn-visualizer/issues/1152.

### CODE REVIEW

File: tests/remoteFolderSelector.spec.tsx [hunk 1]
```diff
@@ -352,7 +410,21 @@ it('displays appropriate connection count information', () => {
...
+    if (selected) {
+        window.localStorage.setItem(LOCAL_STORAGE_KEY_SELECTED, JSON.stringify(selected));
+    }
+};
+
+// TODO: Add more tests to cover remaining functionality and edge cases
+// ❌ No test for clicking Edit button
+// ❌ No test for clicking Remove button
+// ❌ No test for Profiler report selection (only performance)
+// ❌ No test verifying error messages display
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply tests/remoteFolderSelector.spec.tsx:431]: Ignoring.

File: tests/remoteFolderSelector.spec.tsx [hunk 2]
```diff
@@ -352,7 +410,21 @@ it('displays appropriate connection count information', () => {
...
+    expect(getButtonWithText(REMOVE_NEW_CONNECTION)).toHaveProperty(HTML_DISABLED, false);
+    expect(getButtonWithText(FETCH_REMOTE_FOLDERS)).toHaveProperty(HTML_DISABLED, false);
 });
+
+const setupConnection = (connection: RemoteConnection[], selected: RemoteConnection = connection?.[0]) => {
+    window.localStorage.setItem(LOCAL_STORAGE_KEY_CONNECTIONS, JSON.stringify(connection));
+
+    if (selected) {
+        window.localStorage.setItem(LOCAL_STORAGE_KEY_SELECTED, JSON.stringify(selected));
+    }
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply tests/remoteFolderSelector.spec.tsx]: Refactored.

File: tests/helpers/getAllButtonsWithText.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import { screen } from '@testing-library/dom';
 
-const getAllButtonsWithText = (text: string) => {
+const getAllButtonsWithText = (text: string): HTMLButtonElement[] => {
     return screen.getAllByRole('button', { name: new RegExp(text, 'i') });
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply tests/helpers/getAllButtonsWithText.tsx:8]: Changed to HTMLElement.

File: tests/remoteFolderSelector.spec.tsx [hunk 4]
```diff
@@ -222,36 +232,85 @@ it('handles API errors gracefully', () => {
...
+    const { reportName } = selectedReport;
 
     screen.getByText(reportName).click();
 
-    const syncButtons = getAllButtonsWithText('Sync remote folder');
-    expect(syncButtons).to.have.length(2);
+    // Wait for the button text to update after selection
+    await waitFor(() => {
+        const toastFilename = screen.queryByTestId(TEST_IDS.TOAST_FILENAME);
+        expect(toastFilename?.textContent.includes(reportName)).toBe(true);
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply tests/remoteFolderSelector.spec.tsx]: Updated.

File: src/hooks/useRemote.tsx [hunk 5]
```diff
@@ -98,31 +100,36 @@ const useRemoteConnection = () => {
 
     const persistentState = {
         get savedConnectionList() {
-            return JSON.parse(getAppConfig('remoteConnections') ?? '[]') as RemoteConnection[];
+            const connectionList = safeJsonParse(getAppConfig(LOCAL_STORAGE_KEY_CONNECTIONS), []).filter(
+                isValidConnection,
+            );
+
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply src/hooks/useRemote.tsx]: Changed.

File: tests/remoteFolderSelector.spec.tsx [hunk 6]
```diff
@@ -222,36 +232,85 @@ it('handles API errors gracefully', () => {
...
 
-// TODO: Fix this test
-it.skip('enables sync buttons when folders are selected', async () => {
-    window.localStorage.setItem('remoteConnections', JSON.stringify(remoteConnection));
-    window.localStorage.setItem('selectedConnection', JSON.stringify(remoteConnection[0]));
-    window.localStorage.setItem(
-        `${remoteConnection[0].name} - performanceFolders`,
-        JSON.stringify(mockRemotePerformanceFolderList),
-    );
+it('set active performance report and syncs it on selection', async () => {
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply tests/remoteFolderSelector.spec.tsx]: Changed.

File: src/hooks/useRemote.tsx [hunk 7]
```diff
@@ -98,31 +100,36 @@ const useRemoteConnection = () => {
...
         deleteSavedReportFolders: (connection?: RemoteConnection) => {
             deleteAppConfig(`${connection?.name} - reportFolders`);
         },
         getSavedPerformanceFolders: (connection?: RemoteConnection) =>
-            JSON.parse(getAppConfig(`${connection?.name} - performanceFolders`) ?? '[]') as RemoteFolder[],
+            safeJsonParse(getAppConfig(`${connection?.name} - performanceFolders`), []) as RemoteFolder[],
+
         setSavedPerformanceFolders: (connection: RemoteConnection | undefined, folders: RemoteFolder[]) => {
-            setAppConfig(`${connection?.name} - performanceFolders`, JSON.stringify(folders));
+            setAppConfig(`${connection?.name} - performanceFolders`, safeJsonStringify(folders, '[]'));
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply src/hooks/useRemote.tsx]: Updated.

File: src/hooks/useRemote.tsx [hunk 8]
```diff
@@ -98,31 +100,36 @@ const useRemoteConnection = () => {
...
-            setAppConfig('selectedConnection', JSON.stringify(connection ?? null));
+            setAppConfig(LOCAL_STORAGE_KEY_SELECTED, safeJsonStringify(connection ?? null));
         },
         getSavedReportFolders: (connection?: RemoteConnection) =>
-            JSON.parse(getAppConfig(`${connection?.name} - reportFolders`) ?? '[]') as RemoteFolder[],
+            safeJsonParse(getAppConfig(`${connection?.name} - reportFolders`), []) as RemoteFolder[],
 
         setSavedReportFolders: (connection: RemoteConnection | undefined, folders: RemoteFolder[]) => {
-            setAppConfig(`${connection?.name} - reportFolders`, JSON.stringify(folders));
+            setAppConfig(`${connection?.name} - reportFolders`, safeJsonStringify(folders, '[]'));
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply src/hooks/useRemote.tsx]: Updated.

File: src/hooks/useRemote.tsx [hunk 9]
```diff
@@ -98,31 +100,36 @@ const useRemoteConnection = () => {
...
+            return connectionList as RemoteConnection[];
         },
         set savedConnectionList(connectionList: RemoteConnection[]) {
-            setAppConfig('remoteConnections', JSON.stringify(connectionList));
+            setAppConfig(LOCAL_STORAGE_KEY_CONNECTIONS, safeJsonStringify(connectionList, '[]'));
         },
         get selectedConnection() {
-            const savedSelectedConnection = JSON.parse(getAppConfig('selectedConnection') ?? 'null');
+            const savedSelectedConnection = safeJsonParse(getAppConfig(LOCAL_STORAGE_KEY_SELECTED), null);
             return (savedSelectedConnection ?? this.savedConnectionList[0]) as RemoteConnection | undefined;
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply src/hooks/useRemote.tsx]: Updated.
